tools/topology/topology2/platform/intel/hdmi-generic.conf

```diff
@@ -275,8 +284,11 @@ IncludeByKey.NUM_HDMIS {
 				}
 				Object.PCM.pcm_caps.1 {
 					name $HDMI4_PCM_CAPS
-					formats 'S32_LE,S16_LE'
+					formats 'S32_LE,S16_LE,IEC958_SUBFRAME_LE'
 					channels_max 8
+					rate_min 0
+					rate_max 0
```

Comment:
do we really need those set to zero? it's weird. @ranj063 any comments?

Reply:
I think this is the first and only use of a integer constraint with enumerated valid values, so there's not a lot of examples (in SOF or elsewhere in ALSA). The code to parse was added to alsa-utils by @ranj063 , so I'll let you confirm. I wonder if a comment would be nice to explain why this is needed. As this is the first example, this bit is likely to be copy'n'pasted to many places down the line....

Reply:
@ranj063, I have not tested w/o these set to 0. It just felt natural to 'disable' the rate_min/max when setting the rates individually, let me test it.

Reply:
@ranj063, @plbossart, @kv2019i, the rate_min/max needs to be set to zero for the rates to be picked up. I only dropped it for HDMI1 (hw:0,3) and [rate_min/max **is not set** to 0]:
```
alsa_conformance_test -P hw:0,3 --dev_info_only                    
------DEVICE INFORMATION------
PCM handle name: hw:0,3
PCM type: HW
card: sofhdadsp [sof-hda-dsp]
device: HDMI1 (*) []
stream: PLAYBACK
available channels: 2 4 6 8
available formats: S16_LE S32_LE IEC958_SUBFRAME_LE
rate range: [48000, 48000]
available rates: 48000
period size range: [12, 4096]
buffer size range: [192, 16384]
```
vs the unmodified HDMI2 (hw:0,4) [rate_min/max **is set** to 0]:
```
alsa_conformance_test -P hw:0,4 --dev_info_only
------DEVICE INFORMATION------
PCM handle name: hw:0,4
PCM type: HW
card: sofhdadsp [sof-hda-dsp]
device: HDMI2 (*) []
stream: PLAYBACK
available channels: 2 4 6 8
available formats: S16_LE S32_LE IEC958_SUBFRAME_LE
rate range: [32000, 192000]
available rates: 32000 44100 48000 88200 96000 176400 192000
period size range: [8, 4096]
buffer size range: [128, 16384]
------------------------------
```

Reply:
Thanks for checking @ujfalusi, wondering if we shouldn't update the compiler then? @ranj063 @jsarha comments?

Reply:
@plbossart I dont think we need to update the compiler. The reason @ujfalusi is having to set the min/max to 0 is because the default is set to 48K in the class definition for pcm_caps conf.

Reply:
erm, what? This is just so confusing.... I really don't see what we would want to *require* this rather silly pattern. Why can't we make this simpler for the rest of us?

Reply:
and clearly when 'rates' is set as a list of values, then min/max are irrelevant, aren't they?

Reply:
Model is that base object/class will set the default value which can be inherited and overridden if needed. If not set by base class/object then its always 0.

Reply:
the problem is that the inheritance is irrelevant since we are moving from a max_rate/min_rate representation to  a 'rates' representation.

Reply:
@plbossart the alternative we can use is to have only rates and drop rate_min/max. That way the default rates will be 48K and when needed can be overridden with other supported rates

Reply:
yes, if the default rate_min == rate_max == 48000 then we might as well use rates = 48000 :-)

Reply:
ok here's the PR for it https://github.com/thesofproject/sof/pull/9017